src/_cffi_src/build_openssl.py

```diff
@@ -46,7 +50,15 @@ def _extra_compile_args(platform):
     When we drop support for CRYPTOGRAPHY_OPENSSL_LESS_THAN_110 we can
     revisit this.
     """
-    if platform not in ["win32", "hp-ux11", "sunos5"]:
+    # make sure the compiler used supports the flags to be added
+    is_gcc = False
+    if get_default_compiler() == "unix":
+        d = dist.Distribution()
+        cmd = config(d)
+        cmd._check_compiler()
+        is_gcc = cmd.compiler.compiler[0].startswith("gcc")
```

Comment:
```suggestion
        is_gcc = "gcc" in cmd.compiler.compiler[0] or "clang" in cmd.compiler.compiler[0]
```

Just to be safe.

Reply:
Made the change, and hope it passes the CI test that keeps failing on pep8. I do test locally, but what I get as a result is not what the CI is reporting - and getting really really tiresome to guess what next.

(tox) devpi@x064:[/data/prj/python/git/cryptography]pycodestyle src/_cffi_src/build_openssl.py

Nothing to report (and I do not like the formatting of the requested change, but tools report what tools report)

(tox) devpi@x064:[/data/prj/python/git/cryptography]flake8 src/_cffi_src/build_openssl.py
ksh: flake8:  not found.

Maybe there is additional stuff from "flake8" processing - but I do not know an easy way to do this locally.

Fingers crossed!